src/fields.jl

```diff
@@ -170,6 +170,17 @@ znodes(ϕ::Field{X, Y, Face}) where {X, Y} = reshape(ϕ.grid.zF[1:end-1], 1, 1,
 
 nodes(ϕ) = (xnodes(ϕ), ynodes(ϕ), znodes(ϕ))
 
+function dims(field::Field{LX, LY, LZ}) where {LX, LY, LZ}
+    xdim(LX), ydim(LY), zdim(LZ)
+end
+
+xdim(::Type{Face}) = "xF"
```

Comment:
It's supposed to return the x dimension of a field (either xC or xF). Similarly for ydim and zdim. Should we rename them to something else? I'll move them to the output_writers file in the next commit.

Reply:
Why do you refer to `xC` as the 'dimension' of a field? Maybe this makes sense, I just don't quite understand. Perhaps this is the 'location' of a field?

Note there is a function `location(f)` which returns a tuple of types (eg `(Face, Cell, Cell)`) corresponding to `x, y, z` "location" of a field. This function seems to have a similar function for the `x`, `y`, and `z` directions, except that it translates the types to a string label. Would a better name be `x_location_label(Lx)`, etc?

Reply:
> Why do you refer to xC as the 'dimension' of a field? Maybe this makes sense, I just don't quite understand. Perhaps this is the 'location' of a field?

Think of dimension like an axis here. I call it dimension because that's the convention followed by the netcdf metadata. There's a list of dimensions followed by a list of variables. Each variable, unless it's a scalar, should have at least one dimension from the list of dimensions. See the following netcdf file produced by this outputwriter for an example.

```shell
$ ncdump -h dump.nc

netcdf dump {
dimensions:
	zC = 14 ;
	zF = 14 ;
	xC = 14 ;
	yF = 14 ;
	xF = 14 ;
	yC = 14 ;
	Time = UNLIMITED ; // (370 currently)
variables:
	double zC(zC) ;
	double zF(zF) ;
	double xC(xC) ;
	double yF(yF) ;
	double xF(xF) ;
	double yC(yC) ;
	float Time(Time) ;
	float v(Time, zC, yF, xC) ;
	float S(Time, zC, yC, xC) ;
	float w(Time, zF, yC, xC) ;
	float T(Time, zC, yC, xC) ;
	float u(Time, zC, yC, xF) ;
}
```
This format is powerful because there are many tools built for working with it e.g. python's `xarray`.

> Note there is a function location(f) which returns a tuple of types (eg (Face, Cell, Cell)) corresponding to x, y, z "location" of a field. This function seems to have a similar function for the x, y, and z directions, except that it translates the types to a string label. Would a better name be x_location_label(Lx), etc?

I just want a function that gives `("xF", "yC", "zC")` for `u` and so on. Is there a more efficient way to get it from `location`?

Reply:
Okay, thank you, I think I vaguely understand now (I've worked with NetCDF data before, but its been some time). In summary: the idea is that there is an array called `xC` (for example) whose values correspond to the coordinates along the "dimension" that is labeled `xC` of any field in the dataset?

That explains why the "dimension" `xC` is distinct from the "dimension" `xF` in NetCDF output. In the ordinary usage of the word "dimension", however, this distinction is confusing. The dimension in ordinary terminology is simply "x".

This terminology seems quite specific to the NetCDF output, and doesn't really correspond to anything else in Oceananigans. I think it's probably appropriate to put these functions in `output_writers.jl` and give it some name like `netcdf_spatial_dimensions(f)`.

As for your question about efficiency / code length --- I think you could get this functionality with

```julia
abbrev(L) = string(L)[1]

netcdf_spatial_dimemsions(f::Field{X, Y, Z}) where {X, Y, Z} =
    ("x" * abbrev(X), "y" * abbrev(Y), "z" * abbrev(Z))
```

I get

```julia
julia> using Oceananigans

julia> using Oceananigans: Face, Cell

julia> grid = RegularCartesianGrid((3, 3, 3), (1, 1, 1));

julia> a = Field(Face, Cell, Face, CPU(), grid);

julia> netcdf_spatial_dimemsions(a)
("xF", "yC", "zF")
```

maybe you can come up with a better name for `abbrev` (though it's kind of cute as is).